components/admin_console/admin_definition.jsx

```diff
@@ -2389,6 +2389,21 @@ const AdminDefinition = {
                             it.stateIsFalse('LdapSettings.EnableSync'),
                         ),
                     },
+                    {
+                        type: Constants.SettingsTypes.TYPE_TEXT,
+                        key: 'LdapSettings.GuestFilter',
+                        label: t('admin.ldap.guestFilterTitle'),
+                        label_default: 'Guest Filter:',
+                        help_text: t('admin.ldap.guestFilterFilterDesc'),
+                        help_text_default: '(Optional) Enter an AD/LDAP filter to use when searching for guest objects. Only the users selected by the query will be able to access Mattermost as Guests. Guests are prevented from accessing teams or channels upon logging in until they are assigned a team and at least one channel.',
+                        placeholder: t('admin.ldap.guestFilterEx'),
+                        placeholder_default: 'E.g.: "(objectClass=guests)"',
+                        isDisabled: it.both(
+                            it.stateIsFalse('LdapSettings.Enable'),
+                            it.stateIsFalse('LdapSettings.EnableSync'),
+                        ),
+                        isHidden: (config) => it.isnt(it.licensedForFeature('LDAPGroups')) && !config.ServiceSettings.ExperimentalLdapGroupSync,
```

Comment:
I think this feature is not only related to "LDAP Groups" from my perspective it should be valid for any kind of authentication. For example, you have define in your LDAP who is a guest and who is not, so somebody log in for first time in the application, he get a new account with the data that comes from LDAP, and in this case, the role is got too.

Reply:
This setting was recently removed from the other settings. I will remove it.